test/EventStore.Client.Tests/ConnectionStringTests.cs

```diff
@@ -144,6 +145,42 @@ public class ConnectionStringTests {
 
 #endif
 
+	[Fact]
+	public void should_throw_error_when_tls_ca_file_does_not_exists() {
+		var certificateFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "path", "not", "found");
+
+		var connectionString = $"esdb://admin:changeit@localhost:2113/?tls=true&tlsVerifyCert=true&tlsCAFile={certificateFilePath}";
+
+		var exception = Assert.ThrowsAsync<FileNotFoundException>(
+			() => {
+				EventStoreClientSettings.Create(connectionString);
+				return Task.CompletedTask;
+			}
+		);
+
+		Assert.NotNull(exception);
+		Assert.Equal("Failed to load certificate. File was not found.", exception.Result.Message);
+	}
+
+	[Fact]
+	public void should_throw_exception_when_wrong_format_is_used_for_certificate_file_in_connection_string() {
+		// We are using a file that is not a certificate.
+		string certificateFilePath = Assembly.GetExecutingAssembly().Location;
+
+		var connectionString =
+			$"esdb://admin:changeit@localhost:2113/?tls=true&tlsVerifyCert=true&tlsCAFile={certificateFilePath}";
+
+		var exception = Assert.ThrowsAsync<Exception>(
+			() => {
+				EventStoreClientSettings.Create(connectionString);
+				return Task.CompletedTask;
+			}
+		);
+
+		Assert.NotNull(exception);
+		Assert.Equal("Failed to load certificate. Invalid file format.", exception.Result.Message);
+	}
+
```

Comment:
Now that we introduced a custom exception type `InvalidClientCertificateException` we should be able to consolidate these tests using `MemberData` and simplify like the other tests below.

```csharp
	public static IEnumerable<object?[]> InvalidClientCertificates() {
		yield return new object?[] { Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "path", "not", "found") };
		yield return new object?[] { Assembly.GetExecutingAssembly().Location };
	}

	[Theory]
	[MemberData(nameof(InvalidClientCertificates))]
	public void connection_string_with_invalid_client_certificate_should_throw(string clientCertificatePath) =>
		Assert.Throws<InvalidSchemeException>(
			() => EventStoreClientSettings.Create(
				$"esdb://admin:changeit@localhost:2113/?tls=true&tlsVerifyCert=true&tlsCAFile={clientCertificatePath}"
			)
		);
```

Reply:
Done 👍